website/docs/cdktf/examples-and-guides/refactoring.mdx

```diff
@@ -0,0 +1,255 @@
+
+```shell-session
+...
+refactoring-example-stack-move-1    # aws_s3_bucket.my-bucket will be destroyed
+                                    # (because aws_s3_bucket.my-bucket is not in configuration)
+                                    - resource "aws_s3_bucket" "my-bucket" {
+...
+refactoring-example-stack-move-2    # aws_s3_bucket.my-other-bucket (my-other-bucket) will be created
+...
+```
+
+If we want to preserve the state of the resource we need to move it from one stack to another. For this we need to use [`terraform import`](/cli/commands/import) and [`terraform state rm`](/cli/commands/state/rm).
+
+```shell-session
+# First we need to find the terraform name of the resource in the old stack
+# Run this before removing the resource from the old stack
+$ cat cdktf.out/stacks/refactoring-example-stack-move-1/cdk.tf.json | jq '.resource.aws_s3_bucket'
+{
+  "my-bucket": {
+    "//": {
+      "metadata": {
+        "path": "refactoring-example-stack-move-1/my-bucket",
+        "uniqueId": "my-bucket"
+      }
+    },
+    "bucket_prefix": "my-bucket"
+  }
+}
+
+$ cdktf synth # To ensure our synthesized stack is up to date
+
+# Now we need the name of the resource in the new stack to understand
+# where to move the resource to
+$ cat cdktf.out/stacks/refactoring-example-stack-move-2/cdk.tf.json | jq '.resource.aws_s3_bucket'
+{
+  "my-other-bucket": {
+    "//": {
+      "metadata": {
+        "path": "refactoring-example-stack-move-2/my-other-bucket",
+        "uniqueId": "my-other-bucket"
+      }
+    },
+    "bucket_prefix": "my-other-bucket"
+  }
+}
+
+# Now we need to find the provider-specific id of the resource
+$ cd cdktf.out/stacks/refactoring-example-stack-move-1/
+$ terraform init
+$ terraform state show aws_s3_bucket.my-bucket
+# aws_s3_bucket.my-bucket:
+resource "aws_s3_bucket" "my-bucket" {
+    arn                         = "arn:aws:s3:::my-bucket20221208141249058600000001"
+    bucket                      = "my-bucket20221208141249058600000001"
+    bucket_domain_name          = "my-bucket20221208141249058600000001.s3.amazonaws.com"
+    bucket_prefix               = "my-bucket"
+    bucket_regional_domain_name = "my-bucket20221208141249058600000001.s3.amazonaws.com"
+    force_destroy               = false
+    hosted_zone_id              = "Z3AQBSTGFYJSTF"
+    id                          = "my-bucket20221208141249058600000001"
+    object_lock_enabled         = false
+    region                      = "us-east-1"
+    request_payer               = "BucketOwner"
+    tags_all                    = {}
+
+    grant {
+        id          = "554912fda2704333d162d216be50aefb05562e0bf1709997f1d9417cf46087d5"
+        permissions = [
+            "FULL_CONTROL",
+        ]
+        type        = "CanonicalUser"
+    }
+
+    versioning {
+        enabled    = false
+        mfa_delete = false
+    }
+}
+
+# Now that we have the id ("my-bucket20221208141249058600000001") we can import it
```

Comment:
Hmm, from what I recall, import for different resources use different kinds of IDs (ID vs. ARN vs. something else)? If that's true, we should call it out, but if not, ignore this comment.

Reply:
Maybe copy parts of these Terraform docs (or link to them?):

> ID is dependent on the resource type being imported. For example, for AWS EC2 instances it is the instance ID (i-abcd1234) but for AWS Route53 zones it is the zone ID (Z12ABC4UGMOZ2N). Please reference the provider documentation for details on the ID format. If you're unsure, feel free to just try an ID. If the ID is invalid, you'll just receive an error message.

https://developer.hashicorp.com/terraform/cli/commands/import#usage

Reply:
I copied the start and linked to the rest 👍